Automattic/wp-calypso
# Gutenboarding launch flow: plans grid update to support disabled plan

#### Note to anyone reading this while preparing a release of the Editing Toolkit plugin
This change doesn't require any release notes. We're developing a new feature which is currently hidden behind a flag.

#### Changes proposed in this Pull Request
* When custom domain is selected, free plan should be disabled with a clear message displayed. Read pcbO8q-5j-p2#comment-82 for context.
* Enable navigation to _Choose a domain_ step when clicking on _Pick a domain_ link in Plans grid.
* Mark _Choose a plan_ step as completed when selecting a plan.
* Remove `plans-item` duplicated component introduced at #44582 revert.

#### Testing instructions
* Follow instruction from **Getting started** section of #43750 to have the new launch flow enabled.
* Create a site using [wordpress.com/new](https://wordpress.com/new), sandbox it and click _Launch site_ button in the editor header.
* Keep the subdomain and advance to next step. Here pressing _Pick a domain_ link should take you back to choose a domain.
* Select a custom domain and advance. Free domain should be now disabled. 
* When selecting a plan, sidebar _Choose a plan_ item should be marked as completed.

#### Screenshot
<img width="547" alt="Screenshot 2020-08-07 at 11 09 23" src="https://user-images.githubusercontent.com/14192054/89625760-0c8b2000-d8a1-11ea-8fa4-1a9f51fa2642.png">


Fixes part of #43750
